server/src/main/java/org/elasticsearch/search/SearchPhaseResult.java

```diff
@@ -43,6 +44,10 @@ protected SearchPhaseResult(StreamInput in) throws IOException {
         super(in);
     }
 
+    public boolean hasSearchContext() {
```

Comment:
Can you please provide documentation for this method

Reply:
This was actually defined only on the `QuerySearchResult` class that inherits from `SearchPhaseResults` ([link](https://github.com/elastic/elasticsearch/blob/638a45009cb76059dd914c70eb4aa7e777768da2/server/src/main/java/org/elasticsearch/search/query/QuerySearchResult.java#L362)) I've added this to the parent `SearchPhaseResult` so that I can change the signature of `releaseIrrelevantSearchContext` and make it operate with both `QuerySearchResult` and `RankFeatureResult`.  Will add some documentation through to the base class as suggested :)